plugins/importexport/native/filter/SubmissionFileNativeXmlFilter.inc.php

```diff
@@ -121,12 +120,29 @@ function createSubmissionFileNode($doc, $submissionFile) {
 		}
 
 		$submissionFileNode->appendChild($revisionNode);
-
 		// Embed the file contents
-		$embedNode = $doc->createElementNS($deployment->getNamespace(), 'embed', base64_encode(file_get_contents($submissionFile->getFilePath())));
-		$embedNode->setAttribute('encoding', 'base64');
-		$revisionNode->appendChild($embedNode);
 
+		if (array_key_exists('no-embed', $this->opts)) {
+			$hrefNode = $doc->createElementNS($deployment->getNamespace(), 'href');
+			if (array_key_exists('use-file-urls', $this->opts)) {
+				$app = Application::getApplication();
+				$dispatcher = $app->getDispatcher();
+				$params = ["fileId" => $submissionFile->getFileId(),
+						   "revision" => $submissionFile->getRevision(),
+						   "submissionId" => $submissionFile->getSubmissionId(),
+						   "stageId" => 1];
```

Comment:
`stageId` is the workflow stage ID. You can get this using `SubmissionFileDAO::getWorkflowStageId`.

Reply:
Ok, I think this and the other issues are all resolved now.